test/e2e/testsuites/specs.go

```diff
@@ -310,3 +320,45 @@ func CreateVolumeSnapshotClass(client restclientset.Interface, namespace *v1.Nam
 
 	return tvsc, tvsc.Cleanup
 }
+func VerifySuccessfulReplicaAzVolumeAttachments(pod PodDetails, azDiskClient *azDiskClientSet.Clientset, storageClassParameters map[string]string) {
+	if storageClassParameters["maxShares"] == "" {
+		return
+	}
+	for _, volume := range pod.Volumes {
+		if volume.PersistentVolume != nil {
+			labelSelector := metav1.LabelSelector{MatchLabels: map[string]string{azureconstants.RoleLabel: "Replica", azureconstants.VolumeNameLabel: volume.PersistentVolume.Name}}
```

Comment:
`PersistentVolume.Name == diskName` may not be the case in pre-provisioned volume cases so you may need to get `PersistentVolume` from the api-server and directly get its `pv.Spec.CSI.VolumeHandle`.

Reply:
I think this should still work in the pre provisioned case, since we're looking for the azvolumeattachment label which from what I can tell is set as azVolumeAttachment.Spec.UnderlyingVolume, and is defined as VolumeAttachmentSource.PersistentVolumeName

Reply:
https://github.com/abhisheksinghbaghel/azuredisk-csi-driver/blob/4a34f8cc953254a10f8bbd0977c61bf2a1a8159b/pkg/controller/common.go#L715-L737

We use the disk name derived from the diskURI as the label for the `AzVolumeAttachment`. In dynamically provisioned volumes, disk name is the same as the persistent volume name unless it goes over the length limit, but in pre-provisioned volumes, disk names are usually not equal to the persistent volume name.